src/main/java/htsjdk/variant/variantcontext/writer/VCFWriter.java

```diff
@@ -228,4 +228,10 @@ public void add(final VariantContext context) {
             throw new RuntimeIOException("Unable to write the VCF object to " + getStreamName(), e);
         }
     }
+
+    @Override
+    public void setVcfHeader(VCFHeader header) {
+        this.mHeader = doNotWriteGenotypes ? new VCFHeader(header.getMetaDataInSortedOrder()) : header;
+        this.vcfEncoder = new VCFEncoder(this.mHeader, this.allowMissingFieldsInHeader, this.writeFullFormatField);
```

Comment:
@droazen I've changed it so that `writeHeader()` calls into `setVcfHeader()`.
@cmnbroad Yes that should work for our case, since we don't need to change the header once it's set. When rejecting, do we still allow repeated calls of `writeHeader()` and `setVcfHeader()` (individually, without calling them both) or just simply check if `header` has been set once before?